### DISCUSSION TITLE
Let users choose their preferred language?

### DISCUSSION THREAD

**Discussion by erikjanwestendorp** [umbraco/Umbraco-CMS#14982] [Features and ideas]
Wouldn't it be nice if users could set the backoffice UI language themselves? Currently, the language can be configured per user by an admin, or another user with access to the users section, of Umbraco. For installations with many users, this can be a lot of work. Is it an idea to allow users to set their preferred language in Umbraco themselves? See the image below for an example: 

![image](https://github.com/umbraco/Umbraco-CMS/assets/7831614/50825ded-46f8-4ef9-83e4-cc5e089b4093)

**Comment by justinbotger** [comment]
I think this is a useful addition. It saves some unnecessary hassle if the users can do this themselves.

**Comment by bjarnef** [comment]
Maybe there should be some user audits in that case, e.g. to see if an admin or user changed the language.

**Comment by erikjanwestendorp** [comment]
@bjarnef Thanks for your reply! I think that adding audits is a good idea indeed! 

Additionally, I also consider the following points that need to be taken into account:

- The dropdown may not need to be directly in the overlay because it lacks a save button, having only a close and logout button. I think its functionality should be similar to changing the password or setting up two-factor authentication.
- It's also important to consider which languages a user can choose. Is it all languages or only the languages installed in Umbraco? It's not inconceivable that there are implementations that have made property labels and descriptions multilingual; in that case, you might want to make only those languages available to users.
